scripts/ipnet.js

```diff
@@ -0,0 +1,261 @@
+#!/usr/node/bin/node
+
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2021 Joyent, Inc.
+ */
+
+/*
+ * netmask value indexed by prefix length
+ * Generated with:
+ * var prefix = Array.apply(null, Array(32)).map(function (_, i) {
+ *     return (((0x1ffffffff << (32 - i)) - 0xffffffff) >>> 0);
+ * });
+ */
+var netmask = [
+    0,           // 0
+    2147483648,  // 1
+    3221225472,  // 2
+    3758096384,  // 3
+    4026531840,  // 4
+    4160749568,  // 5
+    4227858432,  // 6
+    4261412864,  // 7
+    4278190080,  // 8
+    4286578688,  // 9
+    4290772992,  // 10
+    4292870144,  // 11
+    4293918720,  // 12
+    4294443008,  // 13
+    4294705152,  // 14
+    4294836224,  // 15
+    4294901760,  // 16
+    4294934528,  // 17
+    4294950912,  // 18
+    4294959104,  // 19
+    4294963200,  // 20
+    4294965248,  // 21
+    4294966272,  // 22
+    4294966784,  // 23
+    4294967040,  // 24
+    4294967168,  // 25
+    4294967232,  // 26
+    4294967264,  // 27
+    4294967280,  // 28
+    4294967288,  // 29
+    4294967292,  // 30
+    4294967294,  // 31
+    4294967295   // 32
+];
+
+/*
+ * prefix length indexed by dotted decimal subnet mask
+ */
+var prefix = {
+    '0.0.0.0': 0,
+    '128.0.0.0': 1,
+    '192.0.0.0': 2,
+    '224.0.0.0': 3,
+    '240.0.0.0': 4,
+    '248.0.0.0': 5,
+    '252.0.0.0': 6,
+    '254.0.0.0': 7,
+    '255.0.0.0': 8,
+    '255.128.0.0': 9,
+    '255.192.0.0': 10,
+    '255.224.0.0': 11,
+    '255.240.0.0': 12,
+    '255.248.0.0': 13,
+    '255.252.0.0': 14,
+    '255.254.0.0': 15,
+    '255.255.0.0': 16,
+    '255.255.128.0': 17,
+    '255.255.192.0': 18,
+    '255.255.224.0': 19,
+    '255.255.240.0': 20,
+    '255.255.248.0': 21,
+    '255.255.252.0': 22,
+    '255.255.254.0': 23,
+    '255.255.255.0': 24,
+    '255.255.255.128': 25,
+    '255.255.255.192': 26,
+    '255.255.255.224': 27,
+    '255.255.255.240': 28,
+    '255.255.255.248': 29,
+    '255.255.255.252': 30,
+    '255.255.255.254': 31,
+    '255.255.255.255': 32
+};
+
+/*
+ * The number of IPs in a given prefix length
+ * Generated with:
+ * var wildcard = Array.apply(null, Array(32)).map(function (_, i) {
+ *     return 2**i;
+ * });
+ */
+var wildcard = [
+    4294967295, //  0
```

Comment:
Keep these decimal, however, as it's actually MORE readable this way.

Reply:
Actually since the wildcard is just the xor of the mask, I think it *does* make sense to also be in hex.